src/serializers/fields.rs

```diff
@@ -242,24 +339,67 @@ impl GeneralFieldsSerializer {
         // we don't both with `used_fields` here because on unions, `to_python(..., mode='json')` is used
```

Comment:
The comment just above here says "we maintain the order of the input dict", that's not correct anymore!